packages/react-components/react-progress/src/components/Progress/useProgressStyles.ts

```diff
@@ -1,33 +1,201 @@
+const useRootStyles = makeStyles({
   root: {
-    // TODO Add default styles for the root element
+    display: 'grid',
+    ...shorthands.gap('8px', '0px'),
+    ...shorthands.overflow('hidden'),
+  },
+});
+
+/**
+ * Styles for the title
+ */
+const useLabelStyles = makeStyles({
+  default: {
+    gridRowStart: '1',
+    ...typographyStyles.body1,
+    color: tokens.colorNeutralForeground1,
+  },
+});
+
+/**
+ * Styles for the description
+ */
+const useDescriptionStyles = makeStyles({
+  default: {
+    gridRowStart: '3',
+    ...typographyStyles.caption1,
+    color: tokens.colorNeutralForeground2,
+  },
+});
+
+/**
+ * Styles for the progress bar
+ */
+const useBarStyles = makeStyles({
+  base: {
+    gridColumnStart: '1',
+    gridRowStart: '2',
+    backgroundColor: tokens.colorCompoundBrandBackground,
+
+    '@media screen and (forced-colors: active)': {
+      backgroundColor: 'Highlight',
+    },
+  },
+  medium: {
+    height: barThicknessValues.medium,
+  },
+  large: {
+    height: barThicknessValues.large,
+  },
+  determinate: {
+    width: `var(${progressCssVars.percentageCssVar})`,
+  },
+  nonZeroDeterminate: {
+    width: `var(${progressCssVars.percentageCssVar})`,
+    transitionProperty: 'width',
+    transitionDuration: '0.3s',
+    transitionTimingFunction: 'ease',
+  },
+  indeterminate: {
+    maxWidth: '33%',
+  },
+  ltr: {
+    position: 'relative',
+    backgroundImage: `linear-gradient(
+      to right,
+      ${tokens.colorNeutralBackground6} 0%,
+      ${tokens.colorCompoundBrandBackground} 50%,
+      ${tokens.colorNeutralBackground6} 100%
+    )`,
+    animationName: IndeterminateProgress,
+    animationDuration: '3s',
+    animationIterationCount: 'infinite',
+  },
+
+  rtl: {
+    animationDirection: 'reverse',
   },
+});
+
+const useTrackStyles = makeStyles({
+  base: {
+    gridRowStart: '2',
+    gridColumnStart: '1',
+    backgroundColor: tokens.colorNeutralBackground6,
 
-  // TODO add additional classes for different states and/or slots
+    '@media screen and (forced-colors: active)': {
+      ...shorthands.borderBottom('1px', 'solid', 'CanvasText'),
+    },
+  },
+  medium: {
+    height: barThicknessValues.medium,
+  },
+  large: {
+    height: barThicknessValues.large,
+  },
 });
 
 /**
  * Apply styling to the Progress slots based on the state
  */
 export const useProgressStyles_unstable = (state: ProgressState): ProgressState => {
-  const styles = useStyles();
-  state.root.className = mergeClasses(progressClassNames.root, styles.root, state.root.className);
+  const { indeterminate = false, thickness = 'medium', percentComplete = 0 } = state;
+  const rootStyles = useRootStyles();
+  const barStyles = useBarStyles();
+  const trackStyles = useTrackStyles();
+  const labelStyles = useLabelStyles();
+  const descriptionStyles = useDescriptionStyles();
+  const { dir } = useFluent();
+
+  const valuePercent = !indeterminate ? Math.min(100, Math.max(0, percentComplete! * 100)) : undefined;
```

Comment:
Or if you follow my suggestions below, you can put this value inline when defining the CSS variable, and don't need the extra `valuePercent` variable.

Reply:
Let's also avoid negative conditions and invert them, for example  `const valuePercent = indeterminate ? undefined : Math.min(100, Math.max(0, percentComplete! * 100));`

Reply:
I agree in general Shift. However, I made a suggestion below to move this inline with where it's used, and then the `!indeterminate` check is moved to the surrounding `if` statement. (Mentioning this again here, since that would mean deleting this line entirely).

https://github.com/microsoft/fluentui/pull/24023/files/f7f3e980c1def054eb02ee3beee6d47cfa4b65d0#r984089281

```ts
  if (state.bar && !indeterminate) {
    state.bar.style = {
      [progressCssVars.percentageCssVar]: Math.min(100, Math.max(0, percentComplete * 100)) + '%',
      ...state.bar.style,
    };
  }
```